SP5WWP/m17-packet/m17-packet-encode.c

```diff
@@ -184,20 +186,20 @@ int main(int argc, char* argv[])
                 }
                 else if(argv[i][1]=='n') //-n - number of bytes in packet
                 {
-                    if(atoi(argv[i+1])>0 && atoi(argv[i+1])<=(800-2))
+                    if(atoi(argv[i+1])>0 && atoi(argv[i+1])<=(825-2))
                         num_bytes=atoi(&argv[i+1][0]);
                     else
                     {
-                        fprintf(stderr, "Number of bytes 0 or exceeding the maximum of 798. Exiting...\n");
+                        fprintf(stderr, "Number of bytes 0 or exceeding the maximum of 823. Exiting...\n");
                         return -1;
                     }
                 }
                 else if(argv[i][1]=='T') //-T - User Text String
                 {
                     if(strlen(&argv[i+1][0])>0)
                     {
-                        memset(text, 0, 800*sizeof(char));
-                        memcpy(text, &argv[i+1][0], strlen(argv[i+1]));
+                        memset(text, 0, 825*sizeof(char));
+                        memcpy(text, &argv[i+1][0], 821*sizeof(char));
```

Comment:
Why not `strlen`?

Reply:
because strlen can exceed the size of the array if the user passes a super large chunk of text.

Reply:
Let's add an `if` to guard that part :)

Reply:
how do you do that inside of memcpy?

Reply:
`memcpy(text, argv[i+1], strlen(argv[i+1])<=821 ? strlen(argv[i+1]) : 821);`?